framework/include/outputs/Checkpoint.h

```diff
@@ -16,16 +16,20 @@
 #include <filesystem>
 
 /**
- * Shortcut for determining what type of autosave this checkpoint is.
- * NONE: Not an autosave checkpoint
- * MODIFIED_EXISTING: We took an existing checkpoint and enabled autosave checks on it.
- * SYSTEM_AUTOSAVE: These checkpoints run in the background and output only when sent a signal.
+ * Enumerated type for determining what type of checkpoint this is.
+ * USER_CREATED: Checkpoint is requested by the user in the input file.
+ * SYSTEM_CREATED: This type of checkpoint is created automatically by the
+ *   system for the purpose of writing checkpoints at regularly scheduled
+ *   walltime intervals or when sent a signal.
+ * USER_AND_SYSTEM_CREATED: Checkpoint is requested by the user in the input
```

Comment:
Yes, a checkpoint object is always created. Do you think "USER_CREATED" would be a better name, since the checkpoint is explicitly requested by the user and the system piggybacks off that object to write wall time and signal checkpoints?